src/testdir/test_cmdline.vim

```diff
@@ -4011,4 +4011,36 @@ func Test_cd_bslash_completion_windows()
   let &shellslash = save_shellslash
 endfunc
 
+func Test_msghistory()
+  " NOTE: 'Messages maintainer:' message must be filtered
+  function! GetMessages() abort
+    return filter(split(execute('messages'), "\n"),
+        \        { _, val -> stridx(val, 'Messages maintainer:') != 0 })
+  endfunction
+
+  " After setting 'msghistory' to 2 and outputting a message 4 times with
+  " :echomsg, is the number of output lines of :messages 2?
+  set msghistory=2
+  echomsg 'foo'
+  echomsg 'bar'
+  echomsg 'baz'
+  echomsg 'foobar'
+  call assert_equal(GetMessages(), ['baz', 'foobar'])
+
+  " When the number of messages is 10 and 'msghistory' is changed to 5, is the
+  " number of output lines of :messages 5?
+  set msghistory=10
+  for num in range(1, 10)
+    echomsg num
+  endfor
+  set msghistory=5
+  call assert_equal(len(GetMessages()), 5)
+
+  " Check empty list
+  set msghistory=0
+  call assert_true(empty(GetMessages()))
```

Comment:
Can you please add this to the option documentation `:h 'msghistory'`
```
Setting it to zero clears the message history.
```

Reply:
Done.